### PR TITLE
[BREAKING] Deprecate passing tuple of columns to sort

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/2178

TODO: I will fix `aggregate` deprecation code here or in #2174 depending what gets merged first.

### CODE REVIEW

File: docs/src/man/sorting.md [hunk 1]
```diff
@@ -127,19 +127,19 @@ julia> last(iris, 4)
 
 Keywords used above include `rev` (to sort in reverse),
 and `by` (to apply a function to values before comparing them).
-Each keyword can either be a single value, or can be a tuple or a vector,
-with values corresponding to individual columns.
+Each keyword can either be a single value, a vector,
+with values corresponding to individual columns or a selector: `All`, `Between`, `Not`, `:`, or `Regex`.
```
Comment (nalimilan) [inline docs/src/man/sorting.md:131]:
```suggestion
Each keyword can either be a single value, a vector with values corresponding to
individual columns, or a selector: `All`, `Between`, `Not`, `:`, or `Regex`.
```

File: src/abstractdataframe/sort.jl [hunk 2]
```diff
@@ -260,7 +260,11 @@ function ordering(df::AbstractDataFrame, cols::AbstractVector, lt, by, rev, orde
 end
 
 #### Convert cols from tuple to Array, if necessary
-ordering(df::AbstractDataFrame, cols::Tuple, args...) = ordering(df, [cols...], args...)
+function ordering(df::AbstractDataFrame, cols::Tuple, args...)
+    Base.depwarn("Passing a tuple $cols of column selectors when sorting data ." *
+                 "frame is deprecated. Pass a vector $([cols...]) instead.", :ordering)
```
Comment (nalimilan) [inline src/abstractdataframe/sort.jl:265]:
```suggestion
    Base.depwarn("Passing a tuple $cols of column selectors when sorting data " *
                 "frame is deprecated. Pass a vector $([cols...]) instead.", :ordering)
```

File: src/abstractdataframe/sort.jl [hunk 3]
```diff
@@ -355,7 +354,7 @@ end
 
 Return a copy of data frame `df` sorted by column(s) `cols`.
 `cols` can be either a `Symbol` or `Integer` column index, or
-a tuple or vector of such indices.
+a vector of such indices, 'All', `Not`, `Between`, `:`, or `Regex`.
```
Comment (nalimilan) [inline src/abstractdataframe/sort.jl:357]:
```suggestion
a vector of such indices, `All`, `Not`, `Between`, `:`, or `Regex`.
```

BTW, is that the standard order in which we list these? The position of `:` isn't very logical I'd say.

File: src/dataframe/sort.jl [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 Sort data frame `df` by column(s) `cols`.
 `cols` can be either a `Symbol` or `Integer` column index, or
-a tuple or vector of such indices.
+a vector of such indices, 'All', `Not`, `Between`, `:`, or `Regex`.
```
Comment (nalimilan) [inline src/dataframe/sort.jl:9]:
```suggestion
a vector of such indices, `All`, `Not`, `Between`, `:`, or `Regex`.
```

File: src/abstractdataframe/sort.jl (hunk 3 above)
Comment (bkamins) [reply src/abstractdataframe/sort.jl:357]: I will move `:` to the front everywhere in this PR.

File: src/abstractdataframe/sort.jl [hunk 5]
```diff
@@ -301,49 +305,44 @@ end
              lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward)
 
 Test whether data frame `df` sorted by column(s) `cols`.
-`cols` can be either a `Symbol` or `Integer` column index, or
-a tuple or vector of such indices.
+`cols` can be either a `Symbol` or `Integer` column index,
+a vector of such indices, `All`, `Not`, `Between`, `:`, or `Regex`.
```
Comment (bkamins) [inline src/abstractdataframe/sort.jl:309]:
```suggestion
a vector of such indices, `:`, `All`, `Not`, `Between`, or `Regex`.
```

File: src/abstractdataframe/sort.jl [hunk 6]
```diff
@@ -301,49 +305,44 @@ end
...
 See other methods for a description of other keyword arguments.
 """
-function Base.issorted(df::AbstractDataFrame, cols_new=[]; cols=[],
+function Base.issorted(df::AbstractDataFrame, cols=[];
                        lt=isless, by=identity, rev=false, order=Forward)
-    if cols != []
-        Base.depwarn("issorted(df, cols=cols) is deprecated, use issorted(df, cols) instead",
-                     :issorted)
-        cols_new = cols
+    if cols isa Union{All, Not, Between, Colon, Regex}
```
Comment (bkamins) [inline src/abstractdataframe/sort.jl:318]:
```suggestion
    if cols isa Union{Colon, All, Not, Between, Regex}
```

File: src/dataframe/sort.jl [hunk 7]
```diff
@@ -69,20 +69,18 @@ julia> sort!(df, (:x, order(:y, rev=true)))
...
                     lt=isless, by=identity, rev=false, order=Forward)
     if !(isa(by, Function) || eltype(by) <: Function)
         msg = "'by' must be a Function or a vector of Functions. Perhaps you wanted 'cols'."
         throw(ArgumentError(msg))
     end
-    if cols != []
-        Base.depwarn("sort!(df, cols=cols) is deprecated, use sort!(df, cols) instead",
-                     :sort!)
-        cols_new = cols
+    if cols isa Union{All, Not, Between, Colon, Regex}
```
Comment (bkamins) [inline src/dataframe/sort.jl:78]:
```suggestion
    if cols isa Union{Colon, All, Not, Between, Regex}
```

File: docs/src/man/sorting.md [hunk 8]
```diff
@@ -127,19 +127,19 @@ julia> last(iris, 4)
 
 Keywords used above include `rev` (to sort in reverse),
 and `by` (to apply a function to values before comparing them).
-Each keyword can either be a single value, or can be a tuple or a vector,
-with values corresponding to individual columns.
+Each keyword can either be a single value, a vector with values corresponding to
+individual columns, or a selector: `:`, `All`, `Between`, `Not`, or `Regex`.
```
Comment (bkamins) [inline docs/src/man/sorting.md:131]:
```suggestion
individual columns, or a selector: `:`, `All`, `Not`, `Between`, or `Regex`.
```

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**